xtype/type.go

```diff
@@ -40,19 +40,48 @@ type Type struct {
 	BasicType     *types.Basic
 }
 
-// StructField returns the type of a struct field.
-func (t Type) StructField(name string) (*Type, bool) {
+// StructField holds the type of a struct field and its name.
+type StructField struct {
+	Name string
+	Type *Type
+}
+
+// StructField returns the type of a struct field and its name upon successful match or nil is not
+// found. This method will fail only if f fold is enabled and there are multiple non-exact matches.
+func (t Type) StructField(name string, fold bool, ignore map[string]struct{}) (*StructField, error) {
 	if !t.Struct {
 		panic("trying to get field of non struct")
 	}
 
+	var foldMatches []*StructField
 	for y := 0; y < t.StructType.NumFields(); y++ {
 		m := t.StructType.Field(y)
+		if _, ignored := ignore[m.Name()]; ignored {
+			continue
+		}
 		if m.Name() == name {
-			return TypeOf(m.Type()), true
+			// exact match takes precedence over fold match
+			return &StructField{Name: m.Name(), Type: TypeOf(m.Type())}, nil
+		}
+		if fold && strings.EqualFold(m.Name(), name) {
+			foldMatches = append(foldMatches, &StructField{Name: m.Name(), Type: TypeOf(m.Type())})
+			// keep going to ensure struct does not have another fold match
+		}
+	}
+
+	switch len(foldMatches) {
+	case 0:
+		return nil, nil
+	case 1:
+		return foldMatches[0], nil
+	default:
+		ambNames := make([]string, 0, len(foldMatches))
+		for _, m := range foldMatches {
+			ambNames = append(ambNames, m.Name)
 		}
+		return nil, fmt.Errorf("value for field %s has ambiguous matches %s, "+
+			"use goverter:map to make an explicit choice or goverter:ignore", name, ambNames)
```

Comment:
Let's make this more explicit.
```suggestion
		return nil, fmt.Errorf(`multiple matching fields found for %s. Possible matches: %s

Explicitly define the mapping via goverter:map. Example:

    goverter:map %s %s

See https://github.com/jmattheis/goverter#struct-field-mapping`, name, strings.Join(ambNames, ", "), name, ambNames[0])
```

Reply:
👍